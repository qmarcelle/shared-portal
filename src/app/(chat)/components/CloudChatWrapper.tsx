'use client';

// CloudChatWrapper injects the Genesys Cloud Messenger SDK and manages Messenger-specific events.
// All script loading, Messenger events, and errors are logged for traceability and debugging.

import { useEffect } from 'react';
import { useChatStore } from '../stores/chatStore';

export default function CloudChatWrapper() {
  const { userData } = useChatStore();

  useEffect(() => {
    // Build chatSettings from env and userData
    window.chatSettings = {
      bootstrapUrl: process.env.NEXT_PUBLIC_GENESYS_BOOTSTRAP_URL || '',
      widgetUrl: process.env.NEXT_PUBLIC_GENESYS_WIDGET_URL || '',
      clickToChatJs: process.env.NEXT_PUBLIC_GENESYS_CLICK_TO_CHAT_JS || '',
      clickToChatEndpoint: process.env.NEXT_PUBLIC_CLICK_TO_CHAT_ENDPOINT || '',
      chatTokenEndpoint: process.env.NEXT_PUBLIC_CHAT_TOKEN_ENDPOINT || '',
      coBrowseEndpoint: process.env.NEXT_PUBLIC_COBROWSE_LICENSE_ENDPOINT || '',
      opsPhone: process.env.NEXT_PUBLIC_OPS_PHONE || '',
      opsPhoneHours: process.env.NEXT_PUBLIC_OPS_HOURS || '',
      ...userData,
    };
    console.debug('[CloudChatWrapper] chatSettings:', window.chatSettings);
  }, [userData]);

  useEffect(() => {
    // Load Genesys Cloud bootstrap and widget scripts
    const bootstrapUrl = window.chatSettings?.bootstrapUrl || '';
    const widgetUrl = window.chatSettings?.widgetUrl || '';
    if (!bootstrapUrl) return;
    const bootstrapScript = document.createElement('script');
    bootstrapScript.src = bootstrapUrl;
    bootstrapScript.async = true;
    bootstrapScript.onload = () => {
      console.debug('[CloudChatWrapper] Genesys bootstrap loaded');
      if (!widgetUrl) return;
      const widgetScript = document.createElement('script');
      widgetScript.src = widgetUrl;
      widgetScript.async = true;
      widgetScript.onload = () => {
        console.debug('[CloudChatWrapper] Genesys widgets loaded');
      };
      widgetScript.onerror = (e) => {
        console.error('[CloudChatWrapper] Failed to load widgets script', e);
      };
      document.body.appendChild(widgetScript);
    };
    bootstrapScript.onerror = (e) => {
      console.error('[CloudChatWrapper] Failed to load bootstrap script', e);
    };
    document.body.appendChild(bootstrapScript);

    // Provide openGenesysChat logic
    window.openGenesysChat = () => {
      if (window.Genesys) {
        try {
          window.Genesys('command', 'Messenger.open');
        } catch (e) {
          console.error(
            '[CloudChatWrapper] Error opening Genesys Messenger',
            e,
          );
        }
      }
    };

    return () => {
      // Cleanup scripts and chatSettings
      document.body.removeChild(bootstrapScript);
      if (widgetUrl) {
        const widgetScript = document.querySelector(
          'script[src="' + widgetUrl + '"]',
        );
        if (widgetScript) document.body.removeChild(widgetScript);
      }
      if (window.chatSettings) delete window.chatSettings;
      if (window.openGenesysChat) delete window.openGenesysChat;
      console.debug('[CloudChatWrapper] Cleanup complete');
    };
  }, [userData]);

  return null;
}
