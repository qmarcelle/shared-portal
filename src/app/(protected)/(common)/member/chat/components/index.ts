/**
 * Component Exports
 * Centralizes all component exports following the new consolidated structure
 */

// Main Components
export { ChatTrigger } from './ChatTrigger';
export { ChatWidget } from './ChatWidget';
export { PlanInfoHeader } from './PlanInfoHeader';

// Error Handling
export { ChatErrorBoundary } from './shared/ChatErrorBoundary';

// Integration Components (to be consolidated)
export { GenesysScripts } from './GenesysScripts';
