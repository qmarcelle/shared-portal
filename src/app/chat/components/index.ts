// Core Components
export * from './core/ChatButton';
export * from './core/ChatErrorBoundary';
export * from './core/ChatHeader';
export * from './core/ChatWidget';

// Business Hours Components
export * from './business-hours/BusinessHoursNotification';

// Eligibility Components
export * from './eligibility/EligibilityCheck';

// Plan Switcher Components will be added here when implemented
// export * from './plan-switcher/PlanSwitcher';
