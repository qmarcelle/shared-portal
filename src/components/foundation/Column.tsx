import { ReactNode } from 'react';
import { IComponent } from '../IComponent';

interface ColumnProps extends IComponent {
  children: ReactNode;
}

export const Column = ({
  className = '',
  children,
  onClick,
  tabIndex,
}: ColumnProps) => {
  return (
    <div
      tabIndex={tabIndex}
      className={`flex flex-col ${className}`.trimEnd()}
      onClick={onClick}
    >
      {children}
    </div>
  );
};
