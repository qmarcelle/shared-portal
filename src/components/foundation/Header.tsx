import { IComponent } from '../IComponent';

interface HeaderProps extends IComponent {
  text: string;
  type?: 'title-1' | 'title-2' | 'title-3';
}

export const Header = ({
  text,
  type = 'title-1',
  className = '',
}: HeaderProps) => {
  if (type == 'title-1') {
    return <h1 className={`${type} ${className}`.trimEnd()}>{text}</h1>;
  } else if (type == 'title-2') {
    return <h2 className={`${type} ${className}`.trimEnd()}>{text}</h2>;
  } else {
    return <h3 className={`${type} ${className}`.trimEnd()}>{text}</h3>;
  }
};
