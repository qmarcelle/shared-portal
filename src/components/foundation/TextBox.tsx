import { IComponent } from '../IComponent';

interface TextBoxProps extends IComponent {
  text: string;
  tabFocus?: number;
  display?: 'block' | 'inline';
  type?: 'title-1' | 'title-2' | 'title-3' | 'body-1' | 'body-2';
}

export const TextBox = ({
  text,
  type = 'body-1',
  tabFocus,
  className = '',
  display = 'block',
}: TextBoxProps) => {
  return (
    <p
      className={`${type} ${className}`.trimEnd()}
      tabIndex={tabFocus}
      style={{ display: display }}
    >
      {text}
    </p>
  );
};
