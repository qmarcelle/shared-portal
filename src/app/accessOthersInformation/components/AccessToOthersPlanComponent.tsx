import { Column } from '@/components/foundation/Column';
import { Spacer } from '@/components/foundation/Spacer';
import { IComponent } from '@/components/IComponent';
import { ReactElement } from 'react';
import { AccessOtherPlanDetails } from '../models/AccessOtherPlanDetails';
import { AccessToOthersPlanItem } from './AccessToOthersPlanItem';

interface AccessToOthersPlanComponentProps extends IComponent {
  accessOtherPlanDetails: AccessOtherPlanDetails[] | null;
  header?: ReactElement;
  subHeader?: ReactElement;
  infoIcon: boolean;
}

export const AccessToOthersPlanComponent = ({
  accessOtherPlanDetails,
  header,
  subHeader,
}: AccessToOthersPlanComponentProps) => {
  return (
    <Column className="flex flex-col">
      {header}
      {subHeader && (
        <div>
          {' '}
          <Spacer size={16} />
          {subHeader}
        </div>
      )}
      <Column className="flex flex-col">
        {accessOtherPlanDetails?.map((item, index) => (
          <AccessToOthersPlanItem
            key={index}
            className="mt-8"
            planDetails={item.otherPlanData}
            name={item.memberName}
            dob={item.dob}
          />
        ))}
      </Column>
    </Column>
  );
};
