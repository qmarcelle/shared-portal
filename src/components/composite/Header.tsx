import { useGroup } from '@/app/providers/GroupProvider';
import { IComponent } from '../IComponent';
import { Column } from '../foundation/Column';
import { SiteHeaderMenuSection } from './SiteHeaderMenuSection';
import { SiteHeaderNavSection } from './SiteHeaderNavSection';

export interface HeaderProps extends IComponent {}

export function Header({ className }: HeaderProps) {
  const { group } = useGroup();

  return (
    <header className={`w-full bg-white shadow-md ${className ?? ''}`}>
      <Column className="app-content py-4">
        <div className="flex justify-between items-center">
          <SiteHeaderNavSection
            parentPages={[]}
            onOpenOverlay={() => {}}
            activeSubNavId={null}
            closeMenuAndSubMenu={() => {}}
          />
          <SiteHeaderMenuSection icon={<></>} items={[]} profiles={[]} />
        </div>
      </Column>
    </header>
  );
}
