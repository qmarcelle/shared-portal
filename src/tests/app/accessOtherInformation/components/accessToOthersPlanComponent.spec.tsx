import { AccessToOthersPlanComponent } from '@/app/accessOthersInformation/components/AccessToOthersPlanComponent';
import { Column } from '@/components/foundation/Column';
import { Header } from '@/components/foundation/Header';
import { TextBox } from '@/components/foundation/TextBox';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';

const renderUI = () => {
  render(
    <AccessToOthersPlanComponent
      header={
        <Column>
          <Header type="title-2" text="Others' Plans" />
        </Column>
      }
      subHeader={
        <Column>
          <TextBox text="Below is the access granted to you to other member's plan information." />
        </Column>
      }
      infoIcon={false}
      accessOtherPlanDetails={[
        {
          memberName: 'Ellie Williams',
          dob: '01/01/1993',
          otherPlanData: [
            {
              planName: 'BlueCross BlueShield of Tennessee',
              subscriber: 'Ellie Williams',
              id: 'ABC1234567890',
              policies: 'Medical, Vision, Dental',
            },
          ],
        },
      ]}
    />,
  );
};

describe('AccessToOthersPlanComponent', () => {
  it('should render the UI correctly', async () => {
    const component = renderUI();
    expect(screen.getByText(/Others' Plans/i)).toBeInTheDocument();
    expect(
      screen.getByText(
        /Below is the access granted to you to other member's plan information./i,
      ),
    ).toBeInTheDocument();
    expect(screen.getByText('Ellie Williams')).toBeInTheDocument();
    expect(screen.getByText('DOB: 01/01/1993')).toBeInTheDocument();
    expect(
      screen.getByText('Bluecross Blueshield Of Tennessee'),
    ).toBeInTheDocument();
    expect(screen.getByText('Subscriber: Ellie Williams')).toBeInTheDocument();
    expect(screen.getByText('ID: ABC1234567890')).toBeInTheDocument();
    expect(
      screen.getByText('Policies: Medical, Vision, Dental'),
    ).toBeInTheDocument();

    expect(component).toMatchSnapshot();
  });
});
