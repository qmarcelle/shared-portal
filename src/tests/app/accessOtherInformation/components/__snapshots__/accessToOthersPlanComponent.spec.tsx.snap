// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccessOnMyPlanComponent should render the UI correctly 1`] = `undefined`;

exports[`AccessToOthersPlanComponent should render the UI correctly 1`] = `undefined`;
